### ISSUE TITLE
update license to LGPLv3

### ISSUE THREAD

**Issue by garlick** [flux-framework/flux-core#1887]
We received approval to license all of flux-framework as LGPL 3, under a new LLNL release ID 76440 (thanks @trws!)

We need to update our top-level COPYING files, DISCLAIMER.LLNS, and source code boiler plates for the new license.

Unless there is a good reason to pause and discuss other forms, my vote would be to go with a simple modification of our current source code boiler plate, rather than a potentially shorter one, and to continue with the practice of skipping the boiler plate on internal (as opposed to installed) header files to reduce the noise.

```
/*****************************************************************************\
 *  Copyright (c) 2014 Lawrence Livermore National Security, LLC.  Produced at
 *  the Lawrence Livermore National Laboratory (cf, AUTHORS, DISCLAIMER.LLNS).
 *  LLNL-CODE-76440 All rights reserved.
 *
 *  This file is part of the Flux resource manager framework.
 *  For details, see https://github.com/flux-framework.
 *
 *  This program is free software; you can redistribute it and/or modify it
 *  under the terms of the GNU Lesser General Public License as published
 *  by the Free Software Foundation; either version 3 of the license,
 *  or (at your option) any later version.
 *
 *  Flux is distributed in the hope that it will be useful, but WITHOUT
 *  ANY WARRANTY; without even the IMPLIED WARRANTY OF MERCHANTABILITY or
 *  FITNESS FOR A PARTICULAR PURPOSE.  See the terms and conditions of the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU Lesser General Public License
 *  along with this program; if not, write to the Free Software Foundation,
 *  Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA.
 *  See also:  http://www.gnu.org/licenses/
\*****************************************************************************/
```

@trws mentioned that he had a script to perform this surgery en masse, which sure beats editing every file, so I'm going to assign this task to @trws :-)

Please be sure to avoid changing the boiler plate on source code that doesn't belong to us, e.g.
* src/common/libev
* src/common/liblsd
* src/common/libtap
* src/common/libtomlc99

and in libutil:
* fdwalk
* mnemonic, mn_wordlist
* sds
* sha1
* sha256
* veb

And let's try to preserve the (c) year in the headers, since that's somewhat useful information at times.  My personal preference is to avoid placing a range of dates in these boiler plates becuase that tends to invite an update each year, which is unnecessary noise in the history.

**Comment by garlick** [comment]
@trws are you working this week?  If not, it's actually no big deal for me to make this change.  (I'll take silence as an indication that you're not around).  Thanks!

**Comment by garlick** [comment]
I'm warming up to these SPDX headers.  They are so nice and concise...

**Comment by garlick** [comment]
If we're transplanting source (which we frequently do), it would seem like we would want to retain a URL for the project of origin, as well as the LLNS release ID.  Also, per the LLNS guidelines, a single-license project can reference the license text rather than a COPYRIGHT file (which seems less problematic for transplants, since the SPDX id references the same text really), So what about
```
# Copyright 2014 Lawrence Livermore National Security, LLC and other
# Flux Project Developers (c.f. AUTHORS, DISCLAIMER.LLNS, COPYING)
# LLNL-CODE-76440
#
# This file is part of the Flux resource manager framework.
# For details, see https://github.com/flux-framework.
#
# SPDX-License-Identifier: LGPL-3.0
```

**Comment by trws** [comment]
I’d be perfectly happy with that.  Agreed on pulling in sources from elsewhere and adding their url of origin.  I’m planning to have a listing of all external source in the COPYING file, but it would be great if we could corral our vendored dependencies somewhere, I’m personally fond of a ./external folder or similar, so it’s easier to track.  That’s probably more surgery than we want to do today, but what do you think about moving that way after the release?


---
Sent from Workspace ONE Boxer<https://whatisworkspaceone.com/boxer>

On December 27, 2018 at 2:36:29 PM PST, Jim Garlick <notifications@github.com> wrote:

If we're transplanting source (which we frequently do), it would seem like we would want to retain a URL for the project of origin, as well as the LLNS release ID. Also, per the LLNS guidelines, a single-license project can reference the license text rather than a COPYRIGHT file (which seems less problematic for transplants, since the SPDX id references the same text really), So what about

# Copyright 2014 Lawrence Livermore National Security, LLC and other
# Flux Project Developers (c.f. AUTHORS, DISCLAIMER.LLNS, COPYING)
# LLNL-CODE-76440
#
# This file is part of the Flux resource manager framework.
# For details, see https://github.com/flux-framework.
#
# SPDX-License-Identifier: LGPL-3.0

**Comment by garlick** [comment]
Fine by me.

And on second thought, the LLNL-CODE reference probably doesn't need to be in there, since the release instructions don't include it in the example.  People can always find that via the orig project reference in the transplant case.

**Comment by garlick** [comment]
> Agreed on pulling in sources from elsewhere and adding their url of origin.   I’m planning to have a listing of all external source in the COPYING file

Oops, I might not have been clear - my main point was just that the SPDX header should include the URL of origin for future transplants of _our_ code.  The top level COPYING file would typically (at least in GNU software) contain the detailed license document, in this case for LGPL-3.0, so IMHO we should continue that.

I thought you were proposing a sort of bill of materials, not moving source around, but we could discuss that  - just not ready to give my unqualified endorsement :-)

**Comment by trws** [comment]
Sorry about that, I conflated two things and was unintelligible.  I agree about having the identifier in our code, and we'd have a COPYING file that calls out to a separate LICENSE file, lists the copyright information for our project, and includes a sort of bill of materials for all external code that's under some other license.

I also was thinking that at some later time we might want to try and corral the external code into more distinctly identifiable areas in the directory structure of the project, but wasn't planning to try and tackle that right now.

**Comment by garlick** [comment]
The Flux contributors ref in the SPDX header in source modules is also probably not needed, since everything so far, that is not 3rd party with its own header, is (c) LLNLs (only).  This might be significant later if we (LLNS) want to reuse source in a project with a different license.   Presumably contributions from other people or institutions would come with headers from their own organizations, although we could ask them to conform to, for example, the Flux reference and the SPDX license identifier.   The more broad authorship statements for the project in README.md and COPYRIGHT.adoc do make sense though.

I guess DISCLAIMER.LLNS is replaced per the new instructions with NOTICE (let's tack on .LLNS though).

So for LLNS Flux source, in summary:
```
/* Copyright 2014 Lawrence Livermore National Security, LLC
 * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
 *
 * This file is part of the Flux resource manager framework.
 * For details, see https://github.com/flux-framework.
 *
 * SPDX-License-Identifier: LGPL-3.0
 */
```

Does this seem sane?

**Comment by garlick** [comment]
Closed by #1901
